AMM/project/scripts/swap-exact-input/src/main.sw

```diff
@@ -0,0 +1,58 @@
+script;
+
+use libraries::{AMM, Exchange};
+use std::block::height;
+
+enum InputError {
+    RouteTooShort: (),
+    SwapAmountsNotExact: (),
+}
+
+enum SwapError {
+    PairExchangeNotRegistered: (ContractId, ContractId),
+}
+
+fn main(
+    amm_contract_address: ContractId,
+    assets: Vec<ContractId>,
+    amounts: Vec<u64>,
+) -> u64 {
+    require(assets.len() >= 2, InputError::RouteTooShort);
+    require(amounts.len() == assets.len(), InputError::SwapAmountsNotExact);
+
+    let amm_contract = abi(AMM, amm_contract_address.into());
+
+    let mut latest_bought = amounts.get(1).unwrap();
```

Comment:
would this work if there are more than 2 amounts? specifically the `.get(1)`

Reply:
If my proposed change (function takes a single `input_amount` parameter, and intermediate input amounts are set to the `latest_bought` from the previous swap) then this can become:

`let mut latest_bought = input_amount;`